x-pack/plugins/security_solution/public/common/links/types.ts

```diff
@@ -154,6 +156,13 @@ export interface LinkItem {
    * Locations where the link is visible in the UI
    */
   visibleIn?: AppDeepLinkLocations[];
+
+  /**
+   * Required UI setting to enable a link.
+   * To enable a link when a boolean UiSetting is true, pass the key as a string.
+   * To enable a link when a specific value is set for a UiSetting, pass an object with key and value.
+   */
+  uiSettingRequired?: string | { key: string; value: string | number | boolean };
```

Comment:
I see there are many value types allowed by the uiSettings config:

https://github.com/elastic/kibana/blob/23dd7cdabf350e54160ee9b3db6d819a778f56d6/packages/core/ui-settings/core-ui-settings-common/src/ui_settings.ts#L16-L26

And the generic type is `unknown`: https://github.com/elastic/kibana/blob/23dd7cdabf350e54160ee9b3db6d819a778f56d6/packages/core/ui-settings/core-ui-settings-common/src/ui_settings.ts#L51

I think it would be better to have also `unknown` to the value here:

```suggestion
  uiSettingRequired?: string | { key: string; value: unknown };
```